### PR TITLE
feat(shared): mongodb

### PR DESCRIPTION
**Pull request by hecttoressau**
Hi I create a function which we can use to create a connection to mongo.
In this case I'll have an user our programmers, so I decided to make tests, but I think it could be better. Could you check that and make me suggestion of how I could make a better test?

Also I really appreciate if you can tell me if I am respecting our architecture

### CODE REVIEW

File: packages/shared/__tests__/setupTest.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    info: jest.fn(),
+    error: jest.fn(),
+    warn: jest.fn(),
+    debug: jest.fn(),
+  },
+}));
+
+// TODO add preffixes use to connect to mongoose
+const preffixes = [
+  'messages',
```
Comment (hecttoressau) [inline packages/shared/__tests__/setupTest.js:13]: Here is the part I don't like. 
My function behavior will throw an error when we give a bad prefix (when we don't have that env_var in our file .env.*) so I just add my prefix. The problem is when someone from other team run this test it won't pass. Any suggestions?

File: .env.example [hunk 2]
```diff
@@ -1,7 +1,15 @@
-# MATH Package
-DEV_MONGO_URI=mongodb://127.0.0.1:27017/booking
-DEV_MONGO_USER=booking-user
-DEV_MONGO_PASS=booking-pass
+PREFFIX_MONGO_AUTH_SOURCE=db where you can authenticate when you don't have any information in your db
+PREFFIX_MONGO_URI=mongodb://127.0.0.1:27017/dbName
+PREFFIX_MONGO_USER=
+PREFFIX_MONGO_PASSWORD=
+// Using mongo shared connection is going to ask for a prefix.
+// this in uppercasewill be your preffix in you env_vars
```
Comment (JoseNoriegaa) [inline .env.example:6]: Can you replace `preffix` with `prefix` (with just an `f`) everywhere you wrote it?

File: packages/shared/src/drivers/mongodb/connection-info.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// Internal dependencies
+const Logger = require('../../utils/logger');
+
+const showConnectionInfo = (connection) => {
+  connection.on('error', (err) => {
+    Logger.error(`[mongoconnection]: Connection error event ${err.message}`);
+    process.exit(1);
+  });
+
+  connection.once('open', () => Logger.info('[mongodb]: Connection oppened'));
```
Comment (JoseNoriegaa) [inline packages/shared/src/drivers/mongodb/connection-info.js:10]: typo: `openned`

Can you replace `openned` with `opened`?

File: packages/shared/src/utils/constants.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const supportedEnvs = {
+  PRODUCTION: 'production',
+  DEVELOP: 'develop',
```
Comment (JoseNoriegaa) [inline packages/shared/src/utils/constants.js:3]: Can you change "develop" to "development"?
```suggestion
  DEVELOPMENT: 'development',
```

File: packages/shared/config/environment.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+// Internal dependencies
+const { supportedEnvs } = require('../src/utils/constants');
+
+let path;
+const filename = '.env.';
+
+if (process.env.NODE_ENV === supportedEnvs.PRODUCTION) {
+  path = `/${filename}${process.env.NODE_ENV}`;
+} else {
+  path = join(__dirname, '../../..', filename);
```
Comment (hecttoressau) [inline packages/shared/config/environment.js:13]: when we are not using production we our file will be **.env.** I think it should be .env

Comment (JoseNoriegaa) [reply packages/shared/config/environment.js:13]: That's right, the value of variable `path` (on line 13) will be `<monorepo_directory>/.env` on dev and test environment :)

**Review by JoseNoriegaa — APPROVED** [review]

**Pull request merged**